src/main/java/com/google/api/codegen/transformer/OutputTransformer.java

```diff
@@ -211,51 +222,104 @@ private static int identifier(String s, int startFrom) {
       OutputSpec.LoopStatement loop,
       MethodContext context,
       SampleValueSet valueSet,
-      Map<String, TypeModel> localVars) {
+      ScopeTable localVars) {
 
+    ScopeTable scope = localVars.newChild();
     OutputView.VariableView accessor =
-        accessor(loop.getCollection(), context, valueSet, localVars, loop.getVariable());
+        accessorNewVariable(loop.getCollection(), context, valueSet, scope, loop.getVariable());
     OutputView.LoopView ret =
         OutputView.LoopView.newBuilder()
             .variableType(
                 context
                     .getNamer()
-                    .getAndSaveTypeName(context.getTypeTable(), localVars.get(loop.getVariable())))
+                    .getAndSaveTypeName(context.getTypeTable(), scope.get(loop.getVariable())))
             .variableName(context.getNamer().localVarName(Name.from(loop.getVariable())))
             .collection(accessor)
             .body(
                 loop.getBodyList()
                     .stream()
-                    .map(body -> toView(body, context, valueSet, localVars))
+                    .map(body -> toView(body, context, valueSet, scope))
                     .collect(ImmutableList.toImmutableList()))
             .build();
 
     // The variable is only visible within the loop, delete it from the table before we return.
-    localVars.remove(loop.getVariable());
```

Comment:
Remove the comment for this deleted line, too.

Reply:
Done.